src/vm/hints/set.rs

```diff
@@ -139,80 +39,72 @@ pub fn set_add(
-                                if let MaybeRelocatable::Int(ref range_limit) = set_span {
-                                    range_limit
-                                        .to_i32()
-                                        .ok_or(VirtualMachineError::BigintToUsizeFail)?
-                                } else {
-                                    return Err(VirtualMachineError::ExpectedInteger(set_span));
-                                };
-
-                            for i in (0..range_limit).step_by(elm_size) {
-                                let set_iter = vm
-                                    .memory
-                                    .get_range(&set_ptr.add_usize_mod(i as usize, None), elm_size)
-                                    .map_err(VirtualMachineError::MemoryError)?;
-
-                                if set_iter == elm {
-                                    vm.memory
-                                        .insert(
-                                            &index_addr,
-                                            &MaybeRelocatable::Int(bigint!(i / elm_size as i32)),
-                                        )
-                                        .map_err(VirtualMachineError::MemoryError)?;
-                                    return vm
-                                        .memory
-                                        .insert(
-                                            &is_elm_in_set_addr,
-                                            &MaybeRelocatable::Int(bigint!(1)),
-                                        )
-                                        .map_err(VirtualMachineError::MemoryError);
-                                }
-                            }
-                            return vm
-                                .memory
-                                .insert(&is_elm_in_set_addr, &MaybeRelocatable::Int(bigint!(0)))
-                                .map_err(VirtualMachineError::MemoryError);
-                        }
-                        None => {
-                            return Err(VirtualMachineError::NoRangeCheckBuiltin);
-                        }
-                    }
-                };
+            let _ = vm
+                .builtin_runners
+                .iter()
+                .find(|(name, _)| name.as_str() == "range_check")
+                .ok_or(VirtualMachineError::NoRangeCheckBuiltin)?
+                .1
+                .as_any()
+                .downcast_ref::<RangeCheckBuiltinRunner>()
+                .ok_or(VirtualMachineError::NoRangeCheckBuiltin)?;
+            let elm_size = if let MaybeRelocatable::Int(ref elm_size) = maybe_rel_elm_size {
+                if elm_size.is_zero() {
+                    return Err(VirtualMachineError::ValueNotPositive(elm_size.clone()));
+                }
+                elm_size
+                    .to_usize()
+                    .ok_or_else(|| VirtualMachineError::ValueNotPositive(elm_size.clone()))?
```

Comment:
This could also mean you passed a negative number.

Reply:
I can't find the code but to_usize [returns None](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3ca52cc031a19f887154c56a6f2c57ab) if the given BigInt is negative

Reply:
Exactly. But it also returns `None` if it's greater than `usize::MAX`, [doesn't it](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3ca52cc031a19f887154c56a6f2c57ab)?
The error message is not considering that case.

Reply:
resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/289/commits/4f77a8115240fada313453bcdccd79776a81675b

Reply:
I meant "too big" :facepalm: